### ISSUE TITLE
Enhancement request: add ability to hide private attributes from Variable Explorer Object view

### ISSUE THREAD

**Issue by jbiz25** [spyder-ide/spyder#14198]
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice

## Problem Description

When double clicking an object in the Variable Explorer to open the Object Viewer, one can optionally turn on and off Show Special and Show Callable by using buttons in the Object View window.  This request is to add a third option: "Show private".  Any attrs, properties, and methods with a leading underscore would be hidden when this new option is toggled off.

This would be an improvement to reduce visual clutter for viewing classes that have many private attrs.

## Versions

* Spyder version: 4.2.0

```

**Comment by dalthviz** [comment]
Hi @jbiz25 thanks for the feedback! I think that such filter could be useful. What do you think @spyder-ide/core-developers ?

**Comment by CAM-Gerlach** [comment]
Yeah, I think that would be particularly useful, and I imagine it should be relatively straightforward to implement.

**Comment by ccordoba12** [comment]
Sure, it's a pretty nice suggestion, thanks @jbiz25! We'll try to implement it in a future release.

**Comment by jbiz25** [comment]
HI guys, thanks very much for the consideration!

**Comment by jbiz25** [comment]
Hi @dalthviz @ccordoba12 ...  To add onto this request, I do believe it would be very useful to make it so that when private attrs are being shown, they are placed BELOW the public ones.  As of now, all private attrs are shown at the top.  I build my classes with a fair number of private attrs, and they kind of hide the the public ones on the Variable Explorer.  I am guessing that you alphabetically sort the variables and that this woudl require a tweak so that underscore chars are moved below letters in the sorting algorithm.  Thanks guys!

**Comment by CAM-Gerlach** [comment]
@jbiz25 I'm not sure how likely this is to get implemented anytime soon; I imagine that instead of special-casing `_` we could simply sort all symbols after letters not unsure how generally applicable that change is. For now, is it too much of an inconvenience to simply sort `Name` by ascending instead of descending, so that they get sorted at the end?
